### DISCUSSION TITLE
Ways to identify Heading, subheadings and child heading

### DISCUSSION THREAD

**Discussion by Shubhamkumar782** [pymupdf/pymupdf4llm#182] [Q&A]
Is there a way to identify subheadings and there parent heading as of now I see subheading or child headings are also getting same tag.

**Comment by JorjMcKie** [comment]
No.
There is no big deal magic behind header detection.
All that happens is a full document parsing, counting the number of characters having the same (rounded) font sizes. The most frequent font size (and all smaller ones) is then declared to be body text.
Up to six larger font sizes are mapped to HTML tags `<h1>` to `<h6>`. That's all. No looking at other possible criteria like **bold**, _italic_ or colors and what not else.

There is nothing that comes close to a semantic understanding of what a header "is".
Therefore, you have the option to inject your own wits: you can replace that logic by suppling a callback function.
Please look at the documentation.
